src/core/core.cpp

```diff
@@ -1012,6 +1012,23 @@ void Core::loadFriends()
     delete[] ids;
 }
 
+void Core::loadGroups()
+{
+    QMutexLocker ml{coreLoopLock.get()};
+
+    const size_t groupCnt = tox_conference_get_chatlist_size(tox.get());
+    if (groupCnt == 0) {
+        return;
+    }
+
+    uint32_t* groupIds = new uint32_t[groupCnt];
```

Comment:
I think vector can be used.
```
std::vector<uint32_t> groupIds(groupCount);
tox_conference_get_chatlist(tox.get(), groupIds.data());
for (auto id : groupIds) {
    emit ...
}
```

Why not a raw array? Because you already forgot to deallocate memory :P

Reply:
Or just delete `groupIds`

Reply:
Crap, force pushed the wrong stuff, I already had that fixed^^